Lib/idlelib/debugger.py

```diff
@@ -130,80 +155,187 @@ def make_gui(self):
+                self.buttondata[key][0]['image'] = self.buttondata[key][2]
+                self.buttondata[key][0]['foreground'] = '#aaaaaa'
+                self.buttondata[key][0]['cursor'] = ''
+                self.buttondata[key][0].bind('<1>', 'break')
+                self.buttondata[key][0].bind('<<context-menu>>', 'break')
+            else:
+                self.buttondata[key][0]['image'] = self.buttondata[key][1]
+                self.buttondata[key][0]['foreground'] = '#000000'
+                self.buttondata[key][0].bind('<1>', self.button_cmds[key])
+                self.buttondata[key][0].bind('<<context-menu>>',
+                                             self.button_cmds[key])
+                self.buttondata[key][0]['cursor'] = self.clickable_cursor
+
+    def stack_selection_changed(self):
+        self.show_vars()
+
+    def stack_doubleclick(self):
+        sel = self.stack.selection()
+        if len(sel) == 1:
+            self.show_source(sel[0])
+
+    def stack_contextmenu(self, event):
+        item = self.stack.identify('item', event.x, event.y)
+        if item is not None and item != -1 and item != '':
+            menu = Menu(self.top, tearoff=0)
+            menu.add_command(label='View Source',
+                             command = lambda: self.show_source(item))
+            menu.tk_popup(event.x_root, event.y_root)
+
+    def show_source(self, item):
+        if item in self.framevars:
+            fname = self.framevars[item][2]
+            lineno = self.framevars[item][3]
+            if fname[:1] + fname[-1:] != "<>" and os.path.exists(fname):
+                self.flist.gotofileline(fname, lineno)
+
+    def show_status(self, msg, error=False):
+        self.status['text'] = msg
+        self.status['foreground'] = '#ff0000' if error else '#006600'
+        self.status['font'] = self.status_error_font if error \
+                            else self.status_normal_font
+
+    def clear_stack(self):
+        self.stack.delete(*self.stack.get_children(''))
+        self.vars.delete(*self.vars.get_children(self.locals))
+        self.vars.delete(*self.vars.get_children(self.globals))
+        self.vars.detach(self.locals)
+        self.vars.detach(self.globals)
+        self.var_values = {}
+
+    def add_stackframe(self, frame, lineno, current=False):
+        func = frame.f_code.co_name
+        if func in ("?", "", None):
+            func = '.'
+        try:
+            selfval = frame.f_locals['self']
+        except KeyError:
+            selfval = None;
+        if selfval:
+            # this stackframe represents an object method; preface the method
+            # name with the name of the class
+            if selfval.__class__.__name__ == 'str':
+                # we've got the string representation of the object sent from
+                # the remote debugger; parse out the name of the class, e.g.
+                # from "<random.Random object at 0x...>" extract "Random"
```

Comment:
PEP8 "Comments should be complete sentences. " 
```suggestion
                # We've got the string representation of the object sent from
                # the remote debugger; parse out the name of the class, e.g.
                # from "<random.Random object at 0x...>" extract "Random".
```